accounts/models.py

```diff
@@ -557,3 +656,21 @@ class Meta:
     @classmethod
     def type_from_string(cls, value):
         return cls.type_map.get(value, cls.UNDETERMINED)
+
+
+class UserGDPRDeletionRequest(models.Model):
+    """
+    This model is used to store information about deletion requests received via email and to help comply with GDPR
+    """
+    user = models.ForeignKey(User, null=True, on_delete=models.SET_NULL, related_name='gdpr_deletion_requests')
+    deleted_user = models.ForeignKey(DeletedUser, null=True, on_delete=models.SET_NULL)
+    username = models.CharField(max_length=150, null=True, blank=True)
+    email = models.CharField(max_length=200)
+    date_request_received = models.DateTimeField(auto_now_add=True)
```

Comment:
Would it be useful to also have date fields for the processes that we go through? Maybe we could just have a jsonb field or a char array field for log messages, and just append messages to it

Reply:
You mean to store when `status` changes right? Makes sense, it could just be a text field I guess with `date status` per line. And then have a post save hook in the `UserGDPRDeletionRequest` that will automatically populate that field when `state` changes. If this sounds good I'll go ahead with this idea.

Reply:
yes, right. I'm not sure what the best data type is here. We could have an actual array of messages, instead of manually constructing them, it might be clearer and less prone to errors.
I guess we have something similar with the processing logs.

Reply:
did we do something with this in the end? Open a separate ticket for it?

Reply:
addressed in e2f191ea8e7b657aa533ca7434746989292211d4